source/php/Component/Iframe/Iframe.php

```diff
@@ -4,114 +4,113 @@
-			foreach ($suppliers as $supplier) {
-				$key = array_search($host, $supplier->domain, true);
-
-				if (is_integer($key)) {
-					$this->data['attributeList']['data-supplier-host'] = "//{$supplier->domain[$key]}";
-					$this->data['attributeList']['data-supplier-name'] = $supplier->name;
-					if (isset($supplier->policy)) {
-						$this->data['attributeList']['data-supplier-policy'] = $supplier->policy;
-					}
-				}
-			}
-		}
-
-		return $data;
-	}
+    public function init()
+    {
+        extract($this->data);
+
+        $this->data['classList'][] = 'js-suppressed-iframe';
+
+        if (isset($width)) {
+            $this->data['attributeList']['width'] = $width;
+        }
+
+        if (isset($height)) {
+            $this->data['attributeList']['height'] = $height;
+        }
+
+        if (isset($frameborder)) {
+            $this->data['attributeList']['frameborder'] = $frameborder;
+        }
+
+        if (isset($loading)) {
+            $this->data['attributeList']['loading'] = $loading;
+        }
+
+        $this->data['attributeList']['src'] = "about:blank";
+
+        if (isset($src)) {
+            $srcParsed = parse_url($src);
+            $embedUrl = $srcParsed['scheme'] . '://' . $srcParsed['host'];
+            if (isset($srcParsed['path'])) {
+                $embedUrl .= $srcParsed['path'];
+            }
+            $this->data['attributeList']['data-src'] = $embedUrl;
+
+            $this->setSupplierDataAttributes($embedUrl, $this->data);
+        }
+    }
+    public static function getSuppliers()
+    {
+        $suppliers = array(
+            new Supplier(
+                'Google',
+                array( 'google.com', 'maps.google.com', 'google.se', 'maps.google.se' ),
+                'https://policies.google.com/privacy'
+            ),
+            new Supplier(
+                'YouTube',
+                array( 'youtube.com', 'www.youtube.com', 'youtu.be' ),
+                'https://policies.google.com/privacy'
+            ),
+            new Supplier(
+                'Vimeo',
+                array( 'vimeo.com', 'www.vimeo.com' ),
+                'https://vimeo.com/privacy'
+            ),
+            new Supplier(
+                'Helsingborg Stad',
+                array( 'helsingborg.se', 'www.helsingborg.se' ),
+                'https://helsingborg.se/om-webbplatsen/sa-har-behandlar-vi-dina-personuppgifter/'
+            ),
+            new Supplier(
+                'Mynewsdesk',
+                array( 'helsingborg.mynewsdesk.com', 'mynewsdesk.com' ),
+                'https://www.mynewsdesk.com/se/about/terms-and-conditions/'
+            ),
+            new Supplier(
+                'KommersAnnons.se',
+                array( 'kommersannons.se' ),
+                'https://kommersannons.se/'
+            ),
+
+        );
+
+        if (function_exists('apply_filters')) {
+            $suppliers = apply_filters(__NAMESPACE__ . '\\' . ucfirst(__FUNCTION__), $suppliers);
```

Comment:
May be a candidate for a return here? We always strive to return as soon as possible.